client/src/components/Button.tsx

```diff
@@ -0,0 +1,3 @@
+export default function Button({ action, children }) {
```

Comment:
Great job with the Cart modal - everything is neat so far and works as expected! However, one small comment I have is regarding the component props here and in the `Modal` component. Since we are using TypeScript, maybe it would be good to also show the types of passed props because without it they implicitly have a type of 'any'. [Details on this page.](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#any)